arangod/Cluster/MaintenanceFeature.cpp

```diff
@@ -1290,11 +1291,19 @@ Result MaintenanceFeature::requeueAction(
              action->getState() == ActionState::FAILED);
   if (action->describe().get(NAME) == SYNCHRONIZE_SHARD) {
     increaseNumberOfSyncShardActionsQueued();
+    // We intentionally ignore the result here, since we **must**
+    // count the job anyway, since it will be scheduled. And thus it will
+    // eventually be executed, which will decrease the counter again!
   }
+  // We use a scope guard to prevent miscounting if something throws during
+  // the registration:
+  ScopeGuard scopeGuard(
+      [&]() noexcept { decreaseNumberOfSyncShardActionsQueued(); });
   auto newAction =
       std::make_shared<maintenance::Action>(*this, action->describe());
   newAction->setPriority(newPriority);
   std::unique_lock guard(_actionRegistryLock);
   registerAction(std::move(newAction));
+  scopeGuard.cancel();
```

Comment:
Can we replace this with try...catch since it is clear that only on exception being thrown we will decrese the count